sharding/nodesCoordinator/sovereignIndexHashedNodesCoordinator.go

```diff
@@ -0,0 +1,234 @@
+	if err != nil {
+		return err
+	}
+
+	nodesConfig.nbShards = 1
+	return nil
+}
+
+// ComputeConsensusGroup will generate a list of validators based on the eligible list
+// and each eligible validator weight/chance
+func (ihnc *sovereignIndexHashedNodesCoordinator) ComputeConsensusGroup(
+	randomness []byte,
+	round uint64,
+	shardID uint32,
+	epoch uint32,
+) (validatorsGroup []Validator, err error) {
+	var selector RandomSelector
+	var eligibleList []Validator
+
+	log.Trace("computing consensus group for",
+		"epoch", epoch,
+		"shardID", shardID,
+		"randomness", randomness,
+		"round", round)
+
+	if len(randomness) == 0 {
+		return nil, ErrNilRandomness
+	}
+
+	ihnc.mutNodesConfig.RLock()
+	nodesConfig, ok := ihnc.nodesConfig[epoch]
+	if ok {
+		if shardID != core.SovereignChainShardId {
+			log.Warn("shardID is not ok, expected a sovereign chain id", "shardID", shardID, "nbShards", nodesConfig.nbShards)
+			ihnc.mutNodesConfig.RUnlock()
+			return nil, ErrInvalidShardId
+		}
+		selector = nodesConfig.selectors[shardID]
+		eligibleList = nodesConfig.eligibleMap[shardID]
+	}
+	ihnc.mutNodesConfig.RUnlock()
+
+	if !ok {
+		return nil, fmt.Errorf("%w epoch=%v", ErrEpochNodesConfigDoesNotExist, epoch)
+	}
+
+	return ihnc.baseComputeConsensusGroup(randomness, round, shardID, epoch, selector, eligibleList)
+}
+
+// GetConsensusValidatorsPublicKeys calculates the validators consensus group for a specific shard, randomness and round number,
+// returning their public keys
+func (ihnc *sovereignIndexHashedNodesCoordinator) GetConsensusValidatorsPublicKeys(
+	randomness []byte,
+	round uint64,
+	shardID uint32,
+	epoch uint32,
+) ([]string, error) {
+	consensusNodes, err := ihnc.ComputeConsensusGroup(randomness, round, shardID, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeys := make([]string, 0)
+	for _, v := range consensusNodes {
+		pubKeys = append(pubKeys, string(v.PubKey()))
+	}
+
+	return pubKeys, nil
+}
+
+// EpochStartPrepare is not implemented for sovereign
+func (ihnc *sovereignIndexHashedNodesCoordinator) EpochStartPrepare(_ data.HeaderHandler, _ data.BodyHandler) {
+	log.Error("sovereignIndexHashedNodesCoordinator.EpochStartPrepare was called, not implemented in sovereign")
+}
+
+func displaySovereignNodesConfiguration(
+	eligible map[uint32][]Validator,
+	waiting map[uint32][]Validator,
+	leaving map[uint32][]Validator,
+	actualRemaining map[uint32][]Validator,
+) {
+	shardID := core.SovereignChainShardId
+	for _, v := range eligible[shardID] {
+		pk := v.PubKey()
+		log.Debug("eligible", "pk", pk, "shardID", shardID)
+	}
+	for _, v := range waiting[shardID] {
+		pk := v.PubKey()
+		log.Debug("waiting", "pk", pk, "shardID", shardID)
+	}
+	for _, v := range leaving[shardID] {
+		pk := v.PubKey()
+		log.Debug("leaving", "pk", pk, "shardID", shardID)
+	}
+	for _, v := range actualRemaining[shardID] {
+		pk := v.PubKey()
+		log.Debug("actually remaining", "pk", pk, "shardID", shardID)
```

Comment:
actual?

Reply:
Copy pasted from original code, but in this case I can change it